src/api/bulk.ts

```diff
@@ -1100,9 +1101,11 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
   readonly #operation: QueryOperation;
   readonly #query: string;
   readonly #pollingOptions: BulkV2PollingOptions;
-  #queryResults: Record[] | undefined;
+  #queryResults: Record[][] | undefined;
```

Comment:
This type was wrong, the results from the parsed body (raw csv) is a 2d array of records.